test_pool/pe/operating_system/test_os_c001.c

```diff
@@ -243,28 +293,61 @@ payload(uint32_t num_pe)
               return;
           }
 
-          if(IS_TEST_FAIL(val_get_status(i))) {
-              val_get_test_data(i, &debug_data, &array_index);
-              val_print(ACS_PRINT_INFO, "\n       array_index value 0x%lx", array_index);
-	      val_print(ACS_PRINT_ERR, "\n       Reg compare failed for PE=%d for reg: ", i);
-	      if (array_index >> 32) {
-	          cache_index = (array_index >> 32) - 1;
-		  val_print(ACS_PRINT_ERR, reg_list[0].reg_desc, 0);
-		  val_print(ACS_PRINT_ERR, " and cache index: %d", cache_index);
-		  val_print(ACS_PRINT_ERR, "\n       Primary PE value = 0x%llx", cache_list[i] & (~reg_list[0].reg_mask));
-	      }
-	      else {
-                  val_print(ACS_PRINT_ERR, reg_list[array_index].reg_desc, 0);
-                  val_print(ACS_PRINT_ERR, "\n       Primary PE value = 0x%llx", rd_data_array[array_index] & (~reg_list[array_index].reg_mask));
-	      }
-              val_print(ACS_PRINT_ERR, "\n       Current PE value = 0x%llx", debug_data);
-              return;
+          pe_buffer = g_pe_reg_info + i;
+
+          for (int cache_index = 0; cache_index < MAX_CACHE_LEVEL; cache_index++) {
+             if (!(pe_buffer->cache_status[cache_index]) &&
+                                                    (pe_buffer->pe_cache[cache_index] != 0))
+             {
+                 val_print(ACS_PRINT_INFO, "\n       PE Index: %d", i);
+                 val_print(ACS_PRINT_INFO, ", cache index: %d", cache_index);
+                 val_print(ACS_PRINT_INFO, ", size read: %llx\n",
+                                                             pe_buffer->pe_cache[cache_index]);
+             }
+             else if (pe_buffer->pe_cache[cache_index] != 0)
+             {
+                   val_print(ACS_PRINT_ERR, "\n       PE Index: %d", i);
+                   val_print(ACS_PRINT_ERR, ", cache index: %d", cache_index);
+                   val_print(ACS_PRINT_ERR, ", size read: %llx     FAIL\n",
+                                                             pe_buffer->pe_cache[cache_index]);
+                   val_print(ACS_PRINT_ERR, "          PRIMARY PE VALUE : %llx \n",
+                                              cache_list[cache_index] & (~reg_list[0].reg_mask));
+                   val_print(ACS_PRINT_ERR, "          CURRENT PE VALUE : %llx \n",
```

Comment:
Can we change this message to "Masked Current PE value"?

Reply:
done